projects/isy-angular-widgets/src/lib/incomplete-date/incomplete-date.component.html

```diff
@@ -1,10 +1,10 @@
 <p-inputMask
   [readonly]="readonly"
   [disabled]="disabled"
-  (onInput)="onChange($event.target.value)"
+  (onInput)="onChange($any($event.target).value)"
   (onBlur)="onBlur()"
   (onComplete)="onComplete()"
-  (onKeydown)="onKeydown($event)"
+  (onKeydown)="onKeydown($any($event))"
```

Comment:
doubt. what is the problem and why is any the best solution?

Reply:
Without the $any(...) we get a compile error: Object is of type 'unknown'
The compiler says that the .value is an unknown property
That's the fix i found

Reply:
I guess it should be converted to an Event Object?

Reply:
Do you have any source that this is a best practice fix?

Reply:
No, i don't have a clearly best practice source.

I tried many solutions, but no one worked.
The only one wha has a directly effect on template level was this one.

Reply:
I am unsure on how to come to the conclusion that an any (beeing a last resort) is necessary here.

One solution found by just following the errors would be:

`  (onInput)="onChange(castToValue($event))"`

with

```
  castToValue(event: Event): string {
    return (event.target as HTMLInputElement).value;
  }
```

it is a bit clungy though, so arguably any might be acceptable as suggested in https://stackoverflow.com/questions/45964576/type-casting-within-a-template-in-angular-2


The case is another with the second one that just needs trivial type casting:

`  onKeydown(event: KeyboardEvent): void {
    const keyEvent = event as KeyboardEvent`